### DISCUSSION TITLE
266

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#608] [Comments]
http://jojoldu.tistory.com/266

**Comment by jojoldu** [comment]
작성자: joshua-qa

바로 써봤는데 너무 편합니다 ㅎㅎ 감사합니다~!

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

크으! 감사합니다!! 더 좋은 팁이 생기면 또 남기겠습니다^^

**Comment by jojoldu** [comment]
작성자: 임동규

잘 읽었습니다. 훌륭한 글 감사합니다. ^^

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

헉 실장님 감사합니다 ㅠㅠㅠㅠ.. 충성충성!

**Comment by jojoldu** [comment]
작성자: holyeye

감사합니다.
시원시원하게 잘 정리되어 있어서 쉽게 잘 사용했습니다^^!

**Comment by strawoo** [comment]
글 잘읽었습니다! 
json 응답시 한글이 encode 돼서 보기 불편한데, decode 하는 방법없나요?

**Comment by Lanace** [comment]
안녕하세요. 글 잘 보았습니다. 유용한 기능이 될것같네요!
궁금한점이 세션을 유지하거나 할때는 어떻게 해야되나요?ㅠ
로그인 API를 호출 후에 해야되는지... 아니면 특정 세션에 대해서 값을 바꿔줘야되는지... 잘 모르겠습니다ㅠ

**Comment by jojoldu** [comment]
@strawoo 안녕하세요!
decode 결과 출력은 저도 찾아보는중입니다 ㅠ
방법을 알게되면 바로 포스팅하겠습니다!

**Comment by jojoldu** [comment]
@Lanace 안녕하세요! 
음 세션 유지는 포스트맨을 사용할때는 가능했었나요..?
시도는 안했지만 생각으로는
1) 톰켓 세션(기본세션)
로그인 API를 사용해서 로그인한 사용자의 세션 쿠키값을 가져온다
-> 해당 세션 쿠키값을 사용할 API의 리퀘스트 헤더에 담아서 보낸다
2) 외부 세션 저장소 (레디스, DB, 등)
외부 세션 저장소에 키와 값으로 임시 로그인 쿠키값을 생성한다
-> 해당 임시로그인 쿠키값을  사용할 API의 리퀘스트 헤더에 담아서 보낸다

로 처리할 수 있을것 같습니다.
단지 생각만이구요. 실제로는 직접 해봐야할것같습니다 ㅠ
결국 세션은 Request의 헤더에 담긴 쿠키값(JsseionID, PssesionID 등) 을 보고 로그인여부를 체크하기 때문에 해당 서비스에서 사용하는 세션 쿠키 키가 어떤건지 확인후 로그인된 세션의 값을 넣어서 보내면 될것 같습니다.

**Comment by joont92** [comment]
와 정말 좋은 팁입니다 감사합니다!!

**Comment by mirotic91** [comment]
말씀하신대로 기본세션에서 로그인한 사용자의 세션 쿠키값을 API 요청 헤더에 전송하니 잘 됩니다!
꿀기능 소개 감사드립니다 ㅎㅎ

**Comment by gaerae** [comment]
덕분에 새로운 기능을 알아갑니다.
감사합니다.

**Comment by shinjaeuk** [comment]
유용한 정보 감사합니다. ~

세션을 유지 하기 위해서 쿠키의 세션값을 호출하는 API 마다 반복적으로 복사 붙여 넣기를 하고 있는데요.
세션값을 변수에 저장해서 호출시 변수를 활용하는 식으로 자동으로 할 수 있는 방법이 없을까요?

**Comment by jojoldu** [comment]
@shinjaeuk 
안녕하세요 :)
해당 내용은 아래 링크를 참고해보시면 될것 같습니다.
https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html#using-response-handler-scripts

조금 기다리시면 이 내용으로도 포스팅하겠습니다 ^^;
좋은 질문 감사합니다 :)

**Comment by wdh2100** [comment]
6. Controller로 자동생성
https://blog.jetbrains.com/idea/2018/01/whats-new-in-intellij-idea-2018-1-eap-4/
해당 링크가 올바른 주소가 아니네요.

**Comment by namhokim** [comment]
JetBrain 쪽 URL이 변경된 것 같아요.
https://www.jetbrains.com/help/idea/rest-client-in-intellij-idea-code-editor.html
=> 
https://www.jetbrains.com/help/idea/rest-client-tool-window.html

https://blog.jetbrains.com/idea/2018/01/whats-new-in-intellij-idea-2018-1-eap-4/ ??

**Comment by Mommoo** [comment]
정말 좋네요 좋은 글 감사합니다.

**Comment by koreagpa-jayce** [comment]
@jojoldu
@strawoo
```
GET http://api.mysite.com/members?id=1

> {%
client.log(decodeURIComponent(escape(response.body)));
%}
```
스크립트 하나 만드셔서 아래와 같이 사용하시면 좀 더 편합니다.
```
> decoder.sh

# decoder.sh
# client.log(decodeURIComponent(escape(response.body)));
```
위와같이 하면 Response Handler에서 한글 확인이 가능합니다.
더 좋은 방법 아시면 공유 부탁 드려요.

**Comment by alittlenation** [comment]
감사합니다. 이런건 첨해봐서 신기하기만 합니다.
개발 범위를 좀 넓혀야할거 같습니다ㅜㅜ

**Comment by wlsgussla123** [comment]
👍 👍 👍 👍 👍 👍

**Comment by seunggabi** [comment]
좋은 자료 감사합니다 ㅎㅎ
